internal/observability/logger.go

```diff
@@ -1,13 +1,37 @@
 package observability
 
 import (
+	"context"
 	"fmt"
+	"strconv"
 
 	"github.com/go-logr/logr"
+	"go.opentelemetry.io/otel/trace"
 
 	"github.com/TBD54566975/ftl/internal/log"
 )
 
+func AddSpanToLogger(ctx context.Context, span trace.Span) (context.Context, trace.Span) {
+	ctx = wrapLogger(ctx, span.SpanContext())
+	return ctx, span
+}
+
+func AddSpanContextToLogger(ctx context.Context) context.Context {
+	sc := trace.SpanContextFromContext(ctx)
+	return wrapLogger(ctx, sc)
+}
+func wrapLogger(ctx context.Context, sc trace.SpanContext) context.Context {
+	logger := log.FromContext(ctx)
+	attributes := map[string]string{}
+	if sc.HasSpanID() {
```

Comment:
I guess an open question here is if we should be hard coding DD specific attributes. I don't think it matters for now, but we may need to revisit in future.